src/mscorlib/shared/System/Buffers/TlsOverPerCoreLockedStacksArrayPool.cs

```diff
@@ -22,6 +23,15 @@ namespace System.Buffers
     /// </remarks>
     internal sealed partial class TlsOverPerCoreLockedStacksArrayPool<T> : ArrayPool<T>
     {
+        private const uint TlsTrimAfterTicks = 60 * 1000;                           // Trim after 60 seconds for low/moderate pressure
+        private const uint TlsHighTrimAfterTicks = 10 * 1000;                       // Trim after 10 seconds for high pressure
+        private const uint StackTrimAfterTicks = 60 * 1000;                         // Trim after 60 seconds for low/moderate pressure
+        private const uint StackHighTrimAfterTicks = 10 * 1000;                     // Trim after 10 seconds for high pressure
+        private const uint StackRefreshTicks = StackTrimAfterTicks / 4;             // Time bump after trimming (1/4 trim time)
+        private const int StackLowTrimCount = 1;                                    // Trim one item when pressure is low
+        private const int StackMediumTrimCount = 2;                                 // Trim two items when pressure is moderate
+        private const int StackHighTrimCount = MaxBuffersPerArraySizePerCore;       // Trim all items when pressure is high
+
         // TODO: #7747: "Investigate optimizing ArrayPool heuristics"
         // - Explore caching in TLS more than one array per size per thread, and moving stale buffers to the global queue.
         // - Explore dumping stale buffers from the global queue, similar to PinnableBufferCache (maybe merging them).
```

Comment:
You can delete this item, since it's effectively what your PR is doing.

Reply:
Done